tests/cluster_utils.go

```diff
@@ -0,0 +1,166 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"fmt"
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	restclient "k8s.io/client-go/rest"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	netclient "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned/typed/k8s.cni.cncf.io/v1"
+	netv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+)
+
+var kubeconfig *string
+var restConfig *restclient.Config
+var clientset *kubernetes.Clientset
+var netClient *netclient.K8sCniCncfIoV1Client
+const testNamespace = "test-namespace"
+
+func createTestNamespaces() error {
+	By(fmt.Sprintf("creating %s namespace", testNamespace))
+	_, err := clientset.CoreV1().Namespaces().Create(context.TODO(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testNamespace}}, metav1.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to create test namespace")
+	}
+	return nil
+}
+
+func deleteNamespace(namespace string) error {
+	return clientset.CoreV1().Namespaces().Delete(context.TODO(), namespace, metav1.DeleteOptions{})
```

Comment:
it is actually. `deleteNamespace` is a private function that is consumed by `RemoveTestNamespace`..
`RemoveTestNamespace` does an `Eventually` in order to make sure the namespace is terminated appropriately.

this is why in the private function we don't need the Wrap.

In general, I add Wrap to all helper functions as it increases readability in case it fails there.

Reply:
ah I see.
Thanks for explaining.